### DISCUSSION TITLE
Milvus statefulset pods became COMPLETED and cannot be re-scheduled again.

### DISCUSSION THREAD

**Discussion by tianshihan818** [milvus-io/milvus#33437] [Q&A and General discussion]
The problem background is, one node triggered disk-pressure, k8s evict the pods on this node.
As the milvus component pods are managed by replicaset or statefulset, it is reasonable to create a new pod if one of the replicas has been killed. Like this:
![image](https://github.com/milvus-io/milvus/assets/154961771/2741b6e6-cb5f-4c4e-b606-fc6dbf14f3bc)
![image](https://github.com/milvus-io/milvus/assets/154961771/ec848f40-98e3-44fa-8019-ac84d028d117)

But I find sometimes etcd or minio pods are evicted, its status would be COMPLETED, and its statefulset will not re-schedule it to another idle node or re-create a new pod. And there is not any warning for that.
![image](https://github.com/milvus-io/milvus/assets/154961771/1d318df6-7027-4a0c-890a-6d5c52237516)
![image (7)](https://github.com/milvus-io/milvus/assets/154961771/764db08d-11fa-46fb-af05-7ff8e5ff9974)

As we know, milvus components rely on the stable etcd service, so in this scene, etcd service cannot recover to be healthy autpmatically, would lead to milvus cluster crash finally.

This case is randomly happens, sometimes the pod status will be PENDING, thus can recover to be RUNNING later.
I am not sure what decides to this results, k8s mechanism or milvus statefulset pipeline? When it happens, I need to restart the etcd service manually (such delete the completed pods and let the statefulset create the new pods, or scale down & up the statefulset).

**Comment by LoveEachDay** [comment]
@tianshihan818 Etcd pod requires persistent volume and if your volume is bound to any specific worker node. Then this etcd pod is bound to this worker node.

**Reply by tianshihan818** [reply]
Thanks for your reply! Now I understand why the etcd & minio (these pods need pvc) evicted pods cannot be re-scheduled to other nodes. But actually, I have another question, have you meet the COMPLETED status of milvus component pods before (besides the pulsar init pod)?

I have tried to reproduce this case, drain the specific node and evict the pods on it, most time the status of these pods will be PENDING, later recover to be RUNNING when the specific node back to normal.

![image](https://github.com/milvus-io/milvus/assets/154961771/10e2b4c8-430a-4920-935a-074e9d3c83b0)
![image](https://github.com/milvus-io/milvus/assets/154961771/364c5e76-8baf-45fe-a2d5-fb13008bbf28)
![image](https://github.com/milvus-io/milvus/assets/154961771/69ad3e60-fd98-40b9-b839-8e70e725925f)
![image](https://github.com/milvus-io/milvus/assets/154961771/c65ee4fb-d010-4b9c-a756-667ae40800ea)

**Comment by tianshihan818** [comment]
Test again, and changed to COMPLETED status.

![image](https://github.com/milvus-io/milvus/assets/154961771/80301276-c365-49e9-8622-83dc56285835)

Checked this pod events, seems that this is caused by the failed health check, and the retry times exceed the limited.

![image](https://github.com/milvus-io/milvus/assets/154961771/afa0790b-04b4-424a-883d-8dd310c409c7)

**Comment by tianshihan818** [comment]
Take a long time, these completed pods will be cleared and re-created. So this is not a big problem.
Thanks for your help! I will close then.
